Add tests for interim response handling

Acconut opened the pull request:
During one of the HTTP WG session at the IETF 121 meeting in Dublin, resumable uploads was discussed and the topic of support for interim responses in the current HTTP ecosystem came up. @mnot suggested that the cache-tests could be used to test how proxies handle interim responses.

This PR takes a first stab at this. It implements two tests for the 102 and 103 status codes. I would have liked to test an informational status code that is not yet registered to ensure that proxies don't limit their behavior to specific status codes, but currently Node.js' http module does not support sending arbitrary interim responses (see https://github.com/nodejs/node/issues/27921#issuecomment-2223488316). The tests check that proxies pass interim responses through, but are still able to cache the final response.

So far, support for this is varying:
- Apache and Squid meet the expected behavior
- Nginx passes the interim responses through, but doesn't cache the final response
- Caddy ignores the interim responses entirely
- Varnish responds with a 503 upon receiving an interim response
- Trafficserver ignores the 103 but fails to respond for the 102

I can imagine that some of these failures are due to missing configuration settings. For example, for Nginx it was necessary to enable HTTP/1.1 to the origin. Otherwise it fell back to HTTP/1.0 which does not support interim responses as far as I know.

To accommodate the new tests, the `interim_response` and `expected_interim_responses` configurations have been added. Since the Fetch API itself does not expose interim responses to the caller, they have to be collected using the ability to intercept dispatcher in Node.js' `undici` module, which is the implementation behind Node's Fetch API. See https://undici.nodejs.org/#/docs/api/Dispatcher?id=dispatchercomposeinterceptors-interceptor for more details.

I would be happy for any feedback regarding the tests themselves, as well as the code changes.

mnot commented:
This looks really good, thanks! I'll review more deeply as soon as I have a moment, but for now can you document the new directives in CONTRIBUTING.md?

Acconut commented:
I have updated the documentation in CONTRIBUTING.md as well as the schema in testsuite-schema.json.

I also fixed running the tests in browsers, where the `undici` package is not available. This package is now only imported when interim responses are expected, and all interim-related tests are disabled in browsers.

In addition, I added two more test cases to cover the ideas from https://github.com/http-tests/cache-tests/issues/78. Let me know if you had different intentions with them.

mnot commented:
One last thing - `test-engine/lib/tpl/explain-test.liquid` needs to explain what effect the new directives have on the test (this is shown when you hover over a test).

Ping me if you need help.

Acconut commented:
Good catch! I updated the template and it now shows the interim responses sent by the server as well as the checks performed by the client. Here's an example:

<img width="770" alt="image" src="https://github.com/user-attachments/assets/57f1d521-8357-44ba-a485-bdc70f7d6771" />

Let me know if I should change something about them.

mnot commented:
Thanks!
